### PR TITLE
Mechanized Armor Set Patch

### PR DESCRIPTION
**Pull request by Mint-Vanilla**
## Additions

Describe new functionality added by your code, e.g.

- Made armour compatible and functional for CE
- Made advanced musket compatible and functional for CE

## Reasoning

Why did you choose to implement things this way, e.g.
- I used heavy plate armour from VEF:Medieval as a base and added what I believed would be appropriate for a more advanced set of plate armour

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [X] Game runs without errors
- [X] (For compatibility patches) ...with and without patched mod loaded
- [ ] Playtested a colony (specify how long) (Mod is minor and only adds 1 gun and 2 pieces of armour, I feel it's unessesary to test that long term)

### CODE REVIEW

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+						<ArmorRating_Blunt>5.2</ArmorRating_Blunt>
+					</value>
+				</li>
+
+				
+				<!--Plate Armour-->
+				<li Class="PatchOperationAdd">
+					<xpath>Defs/ThingDef[defName="Mechanized_Plate_Apparel"]/statBases</xpath>
+					<value>
+						<Bulk>100</Bulk>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:38]: These should definitely have *some* worn bulk. Probably 10 - 15 would be appropriate. For reference, Recon Armor has 80 Bulk and 10 WornBulk. If you want to have some compensation for the high bulk, you can give is a bit of added weight carrying capacity--perhaps, 10 - 15 (recon armor gives 50, but also weighs more).

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		<match Class="PatchOperationSequence">
+			<operations>
+
+				
+				<!--Plate Helm-->
+				<li Class="PatchOperationAdd">
+					<xpath>Defs/ThingDef[defName="Mechanized_Helm_Apparel"]/statBases</xpath>
+					<value>
+						<Bulk>100</Bulk>
+						<WornBulk>0</WornBulk>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:17]: The bulk is very high for the helmet, and it should have *some* wornBulk. I've give it 5 bulk and 1 wornBulk.

Additionally, the helmet description mentions some protection from harmful gases. I'd include:

```
  <li Class="PatchOperationAdd">
    <xpath>Defs/ThingDef[defName="Mechanized_Helm_Apparel"]/equippedStatOffsets</xpath>
    <value>
      <SmokeSensitivity>-0.8</SmokeSensitivity>
    </value>
  </li>
  ```

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+									<AmmoUser>
+										<magazineSize>8</magazineSize>
+										<reloadTime>10</reloadTime>
+										<ammoSet>AmmoSet_FastMusketBall</ammoSet>
+										<reloadOneAtATime>true</reloadOneAtATime>
+									</AmmoUser>
+									<FireModes>
+										<aiAimMode>AimedShot</aiAimMode>
+									</FireModes>
+									<weaponTags>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:132]: You can remove this `<weaponTags>` part, as it's redundant. If it's not included in the operation, it's simply left alone.

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+									<statBases>
+										<WorkToMake>40000</WorkToMake>
+										<Mass>1.25</Mass>
+										<Bulk>3.90</Bulk>
+										<SwayFactor>1.02</SwayFactor>
+										<ShotSpread>0.36</ShotSpread>
+										<SightsEfficiency>0.9</SightsEfficiency>
+										<RangedWeapon_Cooldown>0.85</RangedWeapon_Cooldown>
+									</statBases>
+									<costList>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:108]: You can remove this `<costList>` part, as it's redundant. If it's not included in the operation, it's simply left alone.

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+								<armorPenetrationSharp>4.755</armorPenetrationSharp>
+								<linkedBodyPartsGroup>Muzzle</linkedBodyPartsGroup>
+							</li>	
+							</tools>	
+							</value>
+							</li>
+								<!--Shooting-->						
+								<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+									<defName>Mechanized_Musket</defName>
+									<statBases>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:99]: You can remove this `<WorktoMake>` part, as it's redundant. If it's not included in the operation, it's simply left alone.

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+								<linkedBodyPartsGroup>Barrel</linkedBodyPartsGroup>
+							</li>
+							<li Class="CombatExtended.ToolCE">
+								<label>Bayonet</label>
+								<capacities>
+									<li>Stab</li>
+								</capacities>
+								<power>23</power>
+								<cooldownTime>2.6</cooldownTime>
+								<armorPenetrationSharp>4.755</armorPenetrationSharp>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:90]: This attack needs a `<armorPenetrationBlunt>` assigned to it as well, as sharp attacks get both.

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+				<li Class="PatchOperationReplace">
+					<xpath>Defs/ThingDef[defName="Mechanized_Plate_Apparel"]/statBases/ArmorRating_Sharp</xpath>
+					<value>
+						<ArmorRating_Sharp>7.2</ArmorRating_Sharp>
+					</value>
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>Defs/ThingDef[defName="Mechanized_Plate_Apparel"]/statBases/ArmorRating_Blunt</xpath>
+					<value>
+						<ArmorRating_Blunt>5.5</ArmorRating_Blunt>
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:51]: You can increase this a bit to about 10.8, as that'd be the appropriate value for it being made of steel with this level of sharp armor.

File: Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+					</value>
+				</li>
+
```
Comment (N7Huntsman) [inline Patches/Mechanized Armor Set/Ammo_&_Apparel_Defs.xml:54]: Go ahead and add these two operations to bring it in line with other CE "power" armor. (And confirm that they work, please.)

```
	<li Class="PatchOperationAdd">
		<xpath>Defs/ThingDef[defName="Mechanized_Plate_Apparel"]/apparel/bodyPartGroups</xpath>
		<value>
			<li>Hands</li>
			<li>Feet</li>
		</value>
	</li>

	<li Class="PatchOperationAddModExtension">
		<xpath>Defs/ThingDef[defName="Mechanized_Plate_Apparel"]</xpath>
		<value>
		  <li Class="CombatExtended.PartialArmorExt">
			  <stats>
				  <li>
					<ArmorRating_Sharp>0.60</ArmorRating_Sharp>
					<parts>
						<li>Neck</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Blunt>0.60</ArmorRating_Blunt>
					<parts>
						<li>Neck</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Sharp>0.90</ArmorRating_Sharp>
					<parts>
						<li>Leg</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Blunt>0.90</ArmorRating_Blunt>
					<parts>
						<li>Leg</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Sharp>0.80</ArmorRating_Sharp>
					<parts>
						<li>Arm</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Blunt>0.80</ArmorRating_Blunt>
					<parts>
						<li>Arm</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Sharp>0.60</ArmorRating_Sharp>
					<parts>
						<li>Hand</li>
					</parts>
				  </li>
				  <li>
					<ArmorRating_Blunt>0.60</ArmorRating_Blunt>
					<parts>
						<li>Hand</li>
					</parts>
				  </li>
			  </stats>
		  </li>
		</value>
	</li>
```

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
No errors on load and patch generally looks good. Just some balance tweaks and other adjustments to bring it in line w/ other "power armor" in CE.

**Comment by Mint-Vanilla** [comment]
All of the changes have been made I believe, I'm mildly concerned about that second thing it said I had done, but I'm sure it's just a normal part of commits to this.

**Pull request merged**